### ISSUE TITLE
Word attributes missing in .vrt file

### ISSUE THREAD

**Issue by fbanados** [UAlbertaALTLab/korp-config#1]
Word Attributes missing:
- [x] lemma
- [ ] dependency
- [x] analysis
- [x] gloss

As seen in
<img width="655" alt="Screenshot 2024-07-29 at 12 29 05 PM" src="https://github.com/user-attachments/assets/0c16ffba-4545-4f15-9837-faf90b5b075b">
<img width="660" alt="Screenshot 2024-07-29 at 12 29 37 PM" src="https://github.com/user-attachments/assets/d79d4782-acb0-479a-b99d-072ccf8a3173">
<img width="668" alt="Screenshot 2024-07-29 at 12 29 47 PM" src="https://github.com/user-attachments/assets/924a2538-7361-4c92-97b7-65064e1e3437">

**Comment by fbanados** [comment]
Fields are currently being presented in two `msd` fields:
<img width="866" alt="Screenshot 2024-07-29 at 12 33 03 PM" src="https://github.com/user-attachments/assets/302a3b8a-9df4-4c4f-9170-9561cf39dd0f">

**Comment by fbanados** [comment]
Corrected mapping of fields solves most of the issues with word attributes.  However, dependencies are **not** present in the base `vrt` file.

**Comment by fbanados** [comment]
Search criteria need to be well-formed in fields.  I've changed the config to at least make behaviour work in the following cases (recreated, following the PDF):
<img width="1035" alt="Screenshot 2024-07-29 at 1 59 27 PM" src="https://github.com/user-attachments/assets/a20ec3bc-ada0-420e-9926-5898bafe6d1e">
<img width="868" alt="Screenshot 2024-07-29 at 2 16 27 PM" src="https://github.com/user-attachments/assets/a0edec05-e453-4af5-b69d-46be95d5d9e0">
<img width="861" alt="Screenshot 2024-07-29 at 2 18 06 PM" src="https://github.com/user-attachments/assets/f232a110-5a92-44b2-899a-4fa7c0cb9e83">
<img width="875" alt="Screenshot 2024-07-29 at 2 19 16 PM" src="https://github.com/user-attachments/assets/997dce9a-4f32-4065-8bd7-6923014fdd92">
<img width="861" alt="Screenshot 2024-07-29 at 2 20 12 PM" src="https://github.com/user-attachments/assets/93f3e730-5e34-4f4f-a369-8669a315268a">
<img width="863" alt="Screenshot 2024-07-29 at 2 22 08 PM" src="https://github.com/user-attachments/assets/a4dda56e-da78-4716-b4fc-34aaf0b9015a">
<img width="869" alt="Screenshot 2024-07-29 at 2 23 01 PM" src="https://github.com/user-attachments/assets/4062e626-cdd3-4c4a-8489-477caf8c8fac">

**Comment by fbanados** [comment]
To finish replicating the behaviour of the slides, we need to recover the `msd` and `dependency` fields into the `vrt` file. I do not think this is the case now.  I believe that is a linguist problem, the programming part would work now.

**Comment by aarppe** [comment]
This is excellent, as many of the lost features are now back! If this works for the A-W corpus, I can easily run the same analyses for the Bloomfield and Miscellaneous corpora, so that we can add them to our Cree text collection.

The `msd` (Morphosyntactic Description) field was a historical vestige and may have been reinterpreted as the `analysis` field. Not all words had a `dependency` field, though many did.

I can recreate the `*.vrt` file, if you remind me what is the tabular order of the linguistic fields?

**Comment by fbanados** [comment]
currently the order is `word lemma analysis gloss`, let me know where you place the `dependency` field and I'll update the script https://github.com/UAlbertaALTLab/korp-config/blob/main/crk_WolfartAhenakew_encode.sh (note that, if you want to run the script locally, you need to change the path.  I've committed the file I'm currently using locally, of course once this is deployed it would point to the server paths)

**Comment by fbanados** [comment]
My plan is once the new machine is setup, I'd put `korp.altlab.dev` there first for testing.

**Comment by fbanados** [comment]
Although it does not immediately have priority, we could address UAlbertaALTLab/korp-frontend#29 and UAlbertaALTLab/korp-frontend#24 as well.

**Comment by aarppe** [comment]
How the regular analysis process works is that the various fields get organized as follows, with tabs in between:

| token | lemma | analysis | dependency | Gloss | Rest ... |
|---|---|---|---|---|---| 
| niwâpamâw | wâpamêw | +V+TA+Ind+1Sg+3SgO | @PRED-TA | s/he sees s.o. | ... |

We could add RW or WN semantic classes, etc. Though one feature that would be good to sort out is how can spaces be included in the fields without them being mistaken for field delimiters, which is supposed to be make use of only tabulators. Now, I've replaced spaces with `&#20;` code, and `+` signs with dots.

**Comment by aarppe** [comment]
https://github.com/UAlbertaALTLab/korp-frontend/issues/24 would be the higher priority.

**Comment by fbanados** [comment]
> How the regular analysis process works is that the various fields get organized as follows, with tabs in between:
> 
> token	lemma	analysis	dependency	Gloss	Rest ...
> niwâpamâw	wâpamêw	+V+TA+Ind+1Sg+3SgO	@PRED-TA	s/he sees s.o.	...

That order is ok!

> We could add RW or WN semantic classes, etc. Though one feature that would be good to sort out is how can spaces be included in the fields without them being mistaken for field delimiters, which is supposed to be make use of only tabulators. Now, I've replaced spaces with `&#20;` code, and `+` signs with dots.

I'm looking into that, as the frontend discusses multiple-value fields which would be what we want for fields like RW/WN/etc.

**Comment by fbanados** [comment]
> [UAlbertaALTLab/korp-frontend#24](https://github.com/UAlbertaALTLab/korp-frontend/issues/24) would be the higher priority.

It should not take me too long to create the config files for a new corpus, but the `vrt` files in the private repo only have the token field.  I can use that if it's sufficient, but maybe we'd want to also have other fields? (`analysis`, `gloss`, `lemma`, etc.)

**Comment by aarppe** [comment]
Yeah, the point is that we have a skeleton *.vrt file with only the tokens and the structural metadata, on which all the layers of linguistic analysis can be added, and rerun when the analyzers improve. The scripts we've got should be able to accomplish the analyses quite quickly. The linguistic analysis fields would be exactly the same as for the Ahenakew-Wolfart corpus. What would differ potentially is the meta-structure of the texts, as the available metadata is different.

**Comment by aarppe** [comment]
There's now a revised version of the analyzed VRT file for the A-W corpus, in: `altlab/crk/generated/ahenakew_wolfart_fst+cg+gloss.vrt`

This is done by the following sequence:

`cat corpora/ahenakew_wolfart.vrt | gawk '{ if(match($0, ".+~$")!=0) sub("~$",""); print; }' | bin/fst-cg-analyze-vrt.sh analyser-gt-strict.hfstol /Users/arppe/gt/lang-crk/src/cg3/disambiguator.cg3 analyser-gt-relaxed.hfstol /Users/arppe/gt/lang-crk/src/cg3/functions.cg3 generator-gt-strict.hfstol | bin/vrt2korp.sh > generated/ahenakew_wolfart_fst+cg+gloss.vrt
`

Some further conversion of special characters to HTML might be needed in the script: `bin/vrt2korp.sh`.

**Comment by aarppe** [comment]
A first version of the Bloomfield corpus in VRT form is presented here: https://github.com/UAlbertaALTLab/korp-frontend/issues/24#issuecomment-2259008306
